### ISSUE TITLE
NoClassDefFoundError GroovyPostbuildAction.java, groovy-postbuild:2.3.1 -> groovy-postbuild:2.4

### ISSUE THREAD

**Issue by niesfisch** [jan-molak/jenkins-build-monitor-plugin#364]
hi,

we recently upgraded the following plugin: **groovy-postbuild:2.3.1 -> groovy-postbuild:2.4**

that leads to the following exception:

`....
href=\"https://jenkins.io/redirect/users-mailing-list\">The users list</a> might be also useful in understanding what has happened.</p><h2>Stack trace</h2><pre style=\"margin:2em; clear:both\">java.lang.NoClassDefFoundError: org/jvnet/hudson/plugins/groovypostbuild/GroovyPostbuildAction\n\tat com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.features.HasBadges.asJson(HasBadges.java:36)\n\tat com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.features.HasBadges.asJson(HasBadges.java:20)\n\tat com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.JobViewSerialiser.serialize(JobViewSerialiser.java:25)\n\tat com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.JobViewSerialiser.serialize(JobViewSerialiser.java:13)\n\tat ...`

after looking at the source of the plugins the situation looks like this:

- the groovy post build plugin was changed and the GroovyPostbuildAction.java is gone
- the HasBadges.java still relies on the Class (https://github.com/jan-molak/jenkins-build-monitor-plugin/blob/13696d6039fdaaebbc494f22e386ea1773a0db74/build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/features/HasBadges.java#L36)
- an alias for xml configs was added (GroovyPostbuildAction.java -> BadgeAction.java), see https://github.com/jenkinsci/groovy-postbuild-plugin/blob/4ac8d3fd88467bc1e4fcf5020b59f1d64475461e/src/main/java/org/jvnet/hudson/plugins/groovypostbuild/GroovyPostbuildDescriptor.java#L93

could you change the code to be compatible with the new version (2.4) of the post build plugin?

our versions: 

- org.jenkins-ci.main:jenkins-war:2.111
- org.jenkins-ci.plugins:build-monitor-plugin:1.12+build.201708172343
- org.jvnet.hudson.plugins:groovy-postbuild:2.4
- org.jenkins-ci.plugins:badge:1.2

cheers and thanks for your efforts
marcel

**Comment by ibiatiroler** [comment]
Hi!

I've filed this issue (also) in https://issues.jenkins-ci.org/browse/JENKINS-50420

BR, Markus

**Comment by dcendents** [comment]
@niesfisch @ibiatiroler 

I did the integration for the badges in the build monitor plugin. I will have a look as soon as I can but it won't be before next week.

Also last time I tried the build (acceptance tests) was broken on master so it might take a while to fix that as well. I'll let you know when I have something working.

**Comment by jmellor** [comment]
This problem seems to have been triggered at my site with the upgrade from Jenkins 2.111 to 2.113

**Comment by matthewpiatt** [comment]
I am also experiencing this issue.

We just upgraded from Jenkins LTS 2.89.3 to 2.107.11.
Build Monitor View version 1.12+build.201708172343 with Groovy Postbuild 2.4 did not work.

However, I tried downgrading both, and Build Monitor View version 1.12+build.201704111018 with Groovy Postbuild 2.3.1 **DOES WORK** for us.

What the user sees when using the bad plugin combo:
![image](https://user-images.githubusercontent.com/2207843/38177405-d3bbf848-35b5-11e8-8a2c-a12602cb2e42.png)

Stack trace:
`
java.lang.NoClassDefFoundError: org/jvnet/hudson/plugins/groovypostbuild/GroovyPostbuildAction
 at com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.features.HasBadges.asJson(HasBadges.java:36)
 at com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.features.HasBadges.asJson(HasBadges.java:20)
 at com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.JobViewSerialiser.serialize(JobViewSerialiser.java:25)
 at com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.JobViewSerialiser.serialize(JobViewSerialiser.java:13)
 at org.codehaus.jackson.map.ser.std.StdContainerSerializers$IndexedListSerializer.serializeContents(StdContainerSerializers.java:122)
 at org.codehaus.jackson.map.ser.std.StdContainerSerializers$IndexedListSerializer.serializeContents(StdContainerSerializers.java:71)
 at org.codehaus.jackson.map.ser.std.AsArraySerializerBase.serialize(AsArraySerializerBase.java:86)
 at org.codehaus.jackson.map.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:446)
 at org.codehaus.jackson.map.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:150)
 at org.codehaus.jackson.map.ser.BeanSerializer.serialize(BeanSerializer.java:112)
 at org.codehaus.jackson.map.ser.StdSerializerProvider._serializeValue(StdSerializerProvider.java:610)
 at org.codehaus.jackson.map.ser.StdSerializerProvider.serializeValue(StdSerializerProvider.java:256)
 at org.codehaus.jackson.map.ObjectMapper._configAndWriteValue(ObjectMapper.java:2575)
 at org.codehaus.jackson.map.ObjectMapper.writeValueAsString(ObjectMapper.java:2097)
 at com.smartcodeltd.jenkinsci.plugins.buildmonitor.api.Respond.withSuccess(Respond.java:15)
 at com.smartcodeltd.jenkinsci.plugins.buildmonitor.BuildMonitorView.fetchJobViews(BuildMonitorView.java:136)
 at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
 at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:343)
 at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:184)
 at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:117)
 at org.kohsuke.stapler.MetaClass$JavaScriptProxyMethodDispatcher.doDispatch(MetaClass.java:473)
 at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
 at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:715)\nCaused: javax.servlet.ServletException
 at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:765)
 at org.kohsuke.stapler.Stapler.invoke(Stapler.java:845)
 at org.kohsuke.stapler.MetaClass$10.dispatch(MetaClass.java:374)
 at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:715)
 at org.kohsuke.stapler.Stapler.invoke(Stapler.java:845)
 at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:782)
 at org.kohsuke.stapler.Stapler.invoke(Stapler.java:845)
 at org.kohsuke.stapler.Stapler.invoke(Stapler.java:649)
 at org.kohsuke.stapler.Stapler.service(Stapler.java:199)
 at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
 at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841)
 at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:154)
 at org.jenkinsci.plugins.ssegateway.Endpoint$SSEListenChannelFilter.doFilter(Endpoint.java:225)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
 at io.jenkins.blueocean.auth.jwt.impl.JwtAuthenticationFilter.doFilter(JwtAuthenticationFilter.java:61)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
 at com.smartcodeltd.jenkinsci.plugin.assetbundler.filters.LessCSS.doFilter(LessCSS.java:47)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
 at io.jenkins.blueocean.ResourceCacheControl.doFilter(ResourceCacheControl.java:134)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
 at javax.servlet.FilterChain$doFilter.call(Unknown Source)
 at com.ceilfors.jenkins.plugins.jiratrigger.ExceptionLoggingFilter.doFilter(ExceptionLoggingFilter.groovy:29)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
 at net.bull.javamelody.MonitoringFilter.doFilter(MonitoringFilter.java:237)
 at net.bull.javamelody.MonitoringFilter.doFilter(MonitoringFilter.java:214)
 at net.bull.javamelody.PluginMonitoringFilter.doFilter(PluginMonitoringFilter.java:88)
 at org.jvnet.hudson.plugins.monitoring.HudsonMonitoringFilter.doFilter(HudsonMonitoringFilter.java:114)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
 at jenkins.metrics.impl.MetricsFilter.doFilter(MetricsFilter.java:125)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
 at hudson.plugins.audit_trail.AuditTrailFilter.doFilter(AuditTrailFilter.java:95)
 at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
 at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:157)
 at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
 at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:64)
 at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
 at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
 at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
 at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
 at jenkins.security.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:117)
 at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
 at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
 at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
 at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)
 at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
 at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)
 at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
 at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:93)
 at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
 at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
 at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)
 at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
 at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:90)
 at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
 at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
 at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
 at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
 at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
 at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
 at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
 at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
 at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
 at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
 at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
 at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
 at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
 at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
 at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
 at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
 at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
 at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
 at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
 at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
 at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
 at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
 at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
 at org.eclipse.jetty.server.Server.handle(Server.java:564)
 at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:317)
 at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
 at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
 at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:110)
 at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
 at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:128)
 at org.eclipse.jetty.util.thread.Invocable$InvocableExecutor.invoke(Invocable.java:222)
 at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:294)
 at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:199)
 at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 at java.lang.Thread.run(Thread.java:748)
`
